src/azure-cli-core/azure/cli/core/parser.py

```diff
@@ -168,7 +172,14 @@ def error(self, message):
         recommender.set_help_examples(self.get_examples(self.prog))
         recommendation = recommender.recommend_a_command()
 
-        az_error = AzCLIError(AzCLIErrorType.ArgumentParseError, message, command=self.prog)
+        az_error = ArgumentParseError(message)
```

Comment:
it seems less reliable to determine error type by text matching in message. Is there way to force message format?

Reply:
Agree. I have beening considering if there is a better way here, too~

Reply:
can we force some message template at where error throw out?

Reply:
No, they are thrown by knack, it's a little hard to customize